src/vss_tools/vspec/vssexporters/vss2samm/vss2samm.py

```diff
@@ -0,0 +1,298 @@
+    "-spl",
+    default=False,
+    show_default=False,
+    help="""\b
+Boolean flag - used to indicate whether to convert VSS specifications in separate ESMF Aspect(s)
+or the whole (selected) VSS specification(s) will be combined into single ESMF Aspect model.
+\033[30m[default: False]\033[0m
+ """,
+)
+@clo.option(
+    "--target-namespace",
+    "-tns",
+    "output_namespace",
+    type=str,
+    default="com.covesa.vss.spec",
+    show_default=False,
+    help="""\b
+Namespace for VSS library, located in specified '--target-folder'.
+Will be used as name of the folder where VSS Aspect models (TTLs) are to be stored.
+This folder will be created as subfolder of the specified '--target-folder' parameter.
+\033[30m[default: com.covesa.vss.spec]\033[0m
+ """,
+)
+@clo.option(
+    "--target-folder",
+    "-tf",
+    type=click.Path(dir_okay=True, file_okay=False, writable=True, path_type=Path),
+    default="vss_ttls",
+    show_default=False,
+    help="""\b
+Path to or name for the target folder, where generated aspect models (.ttl files) will be stored.
+\033[36mNOTE:\033[0m This folder will be created relatively to the folder from which this script is called.
+\033[30m[default: vss_ttls/]\033[0m
+""",
+)
+# END of VSS2SAMM CUSTOM OPTIONS
+@click.command()
+@clo.vspec_opt
+@clo.include_dirs_opt
+@clo.extended_attributes_opt
+@clo.strict_opt
+@clo.aborts_opt
+@clo.uuid_opt
+@clo.expand_opt
+@clo.overlays_opt
+@clo.quantities_opt
+@clo.units_opt
+@clo.types_opt
+@clo.types_output_opt
+@clo.log_level_opt
+@clo.log_file_opt
+def cli(
+    vspec: Path,
+    target_folder: Path,
+    include_dirs: tuple[Path],
+    extended_attributes: tuple[str],
+    strict: bool,
+    aborts: tuple[str],
+    uuid: bool,
+    expand: bool,
+    overlays: tuple[Path],
+    quantities: tuple[Path],
+    units: tuple[Path],
+    types: tuple[Path],
+    types_output: Path,
+    signals_file,
+    output_namespace,
+    split,
+    split_depth,
+    log_level: str,
+    log_file: Path,
+) -> None:
+    """
+    Export as Eclipse Semantic Modeling Framework (ESMF) - Semantic Aspect Meta Model (SAMM) - .ttl files.
+    """
+
+    # TODO: For some reason, the main cli in vspec.py does not seem to get executed
+    #       and log level is always INFO.
+    #       Therefore I use the setLevel of log_level option before further execution of this script.
+    if log_file:
+        file_handler = logging.FileHandler(log_file, mode="w")
+        file_handler.setFormatter(logging.Formatter("%(asctime)s:%(levelname)s:%(message)s"))
+        log.addHandler(file_handler)
+
+    log.setLevel(log_level)
+    # END of copied logging set up from vspec.py::cli
+
+    log.info("Loading VSS Tree...\n")
+
+    tree, datatype_tree = [None, None]
+
+    try:
+        tree, datatype_tree = get_trees(
+            vspec, include_dirs, aborts, strict, extended_attributes, uuid, quantities, units, types, overlays, expand
+        )
+
+    except Exception as e:
```

Comment:
The idea was to add some sort of Exception handling and return a bit more organized response, instead of just printing the whole stack trace in the console.

In fact, this catch is just prompting that something went wrong and also prints the error to the console, before to close the program with error (255).

In our initial implementation, when the tool was build "externally" to vss-tools, we were catching a VSpecError, which was returning enough and meaningful information but after latest refactoring of the vss-tools, I could not find this error, and I could not find other VSS specific errors, which might be thrown by the called functions.

Therefore I went for the general Exception catch.

Reply:
I think the `get_trees()` catches most relevant and expected exceptions, logging it as critical with the original exception text and exiting the program. I really do not see capturing other exceptions here. If we  have exceptions, we have probably undefined behavior somewhere and then I would like to see a stack trace instead.
That might be a bit opinionated  of course so maybe need a  second opinion here

@SebastianSchildt @erikbosch opinions?

Reply:
I remove the try/catch blocks and left it with normal application workflow.